keras_cv/models/segmentation/segment_anything/sam.py

```diff
@@ -0,0 +1,13 @@
+# Copyright 2023 The KerasCV Authors
```

Comment:
I think we should probably have a top-level `SegmentAnything` model which takes an `ImageEncoder` as a backbone and subclasses `Task`. Then the high-level workflows can live on that model.

Then we can also include a preset which includes your ported weights!

Let's also include a reference to the paper and original implementation

Reply:
The image encoder for SAM is a near 1:1 from Detectron2's ViTDet.
IMO it makes sense to have ViTDet as a standalone class/network rather than a SAM encoder only.
That way we can train it from scratch, have it as a standalone object detection model, a backbone for SAM and reuse the same code across all of that

Reply:
I agree @DavidLandup0, I will move the layer to `keras_cv/layers`.

Reply:
> I think we should probably have a top-level `SegmentAnything` model which takes an `ImageEncoder` as a backbone and subclasses `Task`. Then the high-level workflows can live on that model.

That's the plan. I will add a `Task` model in the `kera_cv/models/segmentation/segment_anything/sam.py` file. I am not yet sure how exactly the training step would be implemented with the `Task` API, we could just raise a `NotImplementedError` for now and write a train step as a follow-up.

Reply:
Update: I have moved the image encoder to a standalone backbone. Let me know if that looks good to you. Thanks for the suggestion @DavidLandup0!